sentry-core/src/main/java/io/sentry/core/cache/CacheStrategy.java

```diff
@@ -91,4 +118,192 @@ protected void rotateCacheIfNeeded(final @NotNull File[] files) {
       }
     }
   }
+
+  private void moveInitFlagIfNecessary(
+      final @NotNull File currentFile, final @NotNull File[] notDeletedFiles) {
+    final SentryEnvelope currentEnvelope = readEnvelope(currentFile);
+
+    if (!isValidEnvelope(currentEnvelope)) {
+      return;
+    }
+
+    final Session currentSession = getFirstSession(currentEnvelope);
+
+    if (!isValidSession(currentSession)) {
+      return;
+    }
+
+    // nothing to do if its not true
+    final Boolean currentSessionInit = currentSession.getInit();
+    if (currentSessionInit == null || !currentSessionInit) {
+      return;
+    }
+
+    // we need to move the init flag
+    for (final File notDeletedFile : notDeletedFiles) {
+      final SentryEnvelope envelope = readEnvelope(notDeletedFile);
+
+      if (!isValidEnvelope(envelope)) {
+        continue;
+      }
+
+      SentryEnvelopeItem newSessionItem = null;
+      final Iterator<SentryEnvelopeItem> itemsIterator = envelope.getItems().iterator();
+
+      while (itemsIterator.hasNext()) {
+        final SentryEnvelopeItem envelopeItem = itemsIterator.next();
+
+        if (!isSessionType(envelopeItem)) {
+          continue;
+        }
+
+        final Session session = readSession(envelopeItem);
+
+        if (!isValidSession(session)) {
+          continue;
+        }
+
+        // if its already true, nothing to do
+        final Boolean init = session.getInit();
+        if (init != null && init) {
```

Comment:
L: Maybe we log here? Since we only get valid sessions (status `OK`) at this point, if `init=true` here we have a bug, right? Meaning we didn't close the first one or we're adding `init` to more than one session update

Reply:
makes sense